packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -94,25 +87,39 @@ const ShowHide = (props) => {
 
   const theme = React.useContext(ThemeContext);
   const contentRef = React.useRef(null);
+  const [activeElement, setActiveElement] = React.useState(null);
+  const focusRefClassName = 'show-hide-ref-focusable';
 
   /**
-   * Upon showing hidden content, set focus to the child element if focusRef provided, othervise set focus to the content container.
-   * That would allow assistive technologies to read the newly revealed content.
+   * Upon showing hidden content, set activeElement to focusRef element if provided, othervise set it to content container.
    */
   React.useEffect(() => {
+    let newActiveElement = null;
     if (isOpen) {
-      let element = null;
       if (focusRef?.current) {
-        element = focusRef?.current;
+        if (checkIfAddTabindex(focusRef?.current)) {
+          // not focusable element with no tabindex set, safe to temporary add tabindex that will later be removed on blur
+          focusRef?.current.setAttribute('tabIndex', '-1');
+          focusRef?.current.setAttribute('role', 'group');
+          focusRef?.current.classList.add(focusRefClassName); // needed for onBlur cleanup
```

Comment:
You want to avoid DOM manipulations, where possible, with React.  You should not have to add a class name.  Is this something you can manage with a reference variable?

Reply:
In this case we have no choice but manipulate DOM changing tabindex, and as we have to do so, I prefer the "marker" of it being manipulated being assigned the same way the tabindex was changed in case of asynchronicity between react element and DOM element. What I am trying to avoid is that I set a variable via React then change DOM attribute via setAttribute and at some point React state is not in sync with DOM, giving me a react variable indication it was manipulated when DOM element has already changed the attribute back (or the other way around). I believe that trying to sync a React reference variable with DOM element attribute state change is not 100% possible and the only way to be safe here is consistently attach the marker that tabIndex was changed the same way we are changing tabIndex.

Reply:
I am not understanding how they will get out of sync since React is controlling the rendering.  Maybe I am misunderstanding, but it seems like you just need to keep track of whether you need to remove the attributes.

Reply:
I'll try to rephrase what I meant. Technically, the React element and DOM element aren't the same, that is the exact reason why we don't want to manipulate the DOM elements by "getAttribute" from react - we can't be sure that at some point react element prop like tabIndex is not split second ahead or behind the element.getAttribute('tabindex'). As react is manipulating by virtual DOM rather than read DOM, if we resort to getAttribute tactics I am not sure we should use a react variable to keep track on that.

Reply:
Ok, as it's quite a grey area here, I'll look into adding a reference variable. Will update once have a commit.

Reply:
Variables to track if tabIndex and role were modified added: https://github.com/cerner/terra-core/pull/3838/commits/22a779db64ad198d7388aa83c1d11e1539b14033